src/main/java/net/dv8tion/jda/internal/entities/PrivateChannelImpl.java

```diff
@@ -162,7 +189,9 @@ private void updateUser()
 
     private void checkBot()
     {
-        if (getUser().isBot() && getJDA().getAccountType() == AccountType.BOT)
+        //the only way user is null, is when an event is dispatched that doesn't give us enough information
+        //since events will only happen in this channel if it is between the bot and the user, a null user is a valid channel state
```

Comment:
```suggestion
        //The only way user is null is when an event is dispatched that doesn't give us enough information to build the recipient user,
        // which only happens if this bot sends a message (or otherwise triggers an event) from a shard other than shard 0.
        // The event will be received on shard 0 and not have enough information to build the recipient user.
        //As such, since events will only happen in this channel if it is between the bot and the user, a null user is a valid channel state. 
        // Events cannot happen between a bot and another bot, so the user would never be null in that case.
```